### DISCUSSION TITLE
Make jwt token expiration configurable

### DISCUSSION THREAD

**Discussion by keremkusmezer** [medusajs/medusa#3280] [Feature Requests]
Generated jwt token has a default expiration value of 15 minutes, make it configurable from the settings or app config

**Comment by agustif** [comment]
JWT_EXPIRESIN=
environment variable

**Reply by adevinwild** [reply]
This key does not exist anywhere in the repository unless I am mistaken @agustif 🤔

I find this idea very relevant.
Here is what I could gather as information: 

On the one hand the creation of an admin session expires after 24 hours.
On the other hand for the store, it is done after 30 days.

How do you get 15 minutes?  Is the authentication decoupled from the medusa package (back-end)?

> **References :**
Admin 
https://github.com/medusajs/medusa/blob/6481b0c357c5425d784d0e285fdc6f78061bb4b3/packages/medusa/src/api/routes/admin/auth/create-session.ts#L90

> Store 
https://github.com/medusajs/medusa/blob/6481b0c357c5425d784d0e285fdc6f78061bb4b3/packages/medusa/src/api/routes/store/auth/create-session.ts#L84

**Reply by agustif** [reply]
You're right, sorry for the mix up

**Comment by adevinwild** [comment]
Hi @keremkusmezer 👋

Since  v̶.̶1̶.̶7̶.̶1̶.̶3̶ **v.1.7.6**  it's allowed to tweak the max age of a session cookie : https://github.com/medusajs/medusa/commit/9ebb50104cc1f6c8ef1cea446ae595fb2eb532a2

**Reply by olivermrbl** [reply]
Thanks for pitching in @adevinwild. A minor correction - the options were added in [v1.7.6](https://github.com/medusajs/medusa/releases/tag/v1.7.6) 😊

**Reply by adevinwild** [reply]
> Thanks for pitching in @adevinwild. A minor correction - the options were added in [v1.7.6](https://github.com/medusajs/medusa/releases/tag/v1.7.6) 😊

Oops, thanks for the right version number 😬😊
